npm/pkg/dataplane/policies/policymanager.go

```diff
@@ -0,0 +1,58 @@
+package policies
+
+import "sync"
+
+type PolicyMap struct {
+	sync.Mutex
+	cache map[string]*NPMNetworkPolicy
+}
+
+type PolicyManager struct {
+	policyMap *PolicyMap
+}
+
+func NewPolicyManager() PolicyManager {
+	return PolicyManager{
+		policyMap: &PolicyMap{
+			cache: make(map[string]*NPMNetworkPolicy),
+		},
+	}
+}
+
+func (pMgr *PolicyManager) GetPolicy(name string) (*NPMNetworkPolicy, error) {
+	pMgr.policyMap.Lock()
```

Comment:
I have to read about sync.RWMutex, i do not think we have enough leverage to move to a RWmutex, since most of our operations needs to have write lock. GetPolicy may not even end up in our fnial code. At this moment i anticipate it to be useful in network policy controller.

Reply:
you only need to change the embedded mutex to `sync.RWMutex` and start using `RLock` and `RUnlock` on reads. it's drop-in, no other changes need to be made to the code

Reply:
Okay makes sense, let me switch it up.